classes/security/Validation.inc.php

```diff
@@ -430,6 +430,32 @@ static function canAdminister($administeredUserId, $administratorUserId) {
 		// There were no conflicting roles. Permit administration.
 		return true;
 	}
+
+	/**
+	 * Validation process for imported passwords
+	 * @param $userToImport User ByRef. The user that is being imported. 
+	 * @param $encryption string null, sha1, md5 (or any other encryption algorithm defined)
```

Comment:
This doesn't really use the `$encryption` setting, and there's no indication to the user if the given algorithm wasn't right for the journal -- it'll set a random password instead. I'd prefer if the error condition for a configuration mismatch was shown to the user.

Reply:
The `encryption` parameter checks if an encryption method is being defined. If it does not, then the code manages the `password` as a plain password. If the encryption has a value, then the code checks the `password_needs_rehash` function. Not sure if there is a way to check if the has is sha1 or md5 (is it safe to check the length?)

The code has been changed a little so that the filtering function can return the errors produced from the filtering. (the `Filter` class has an _error array, but I can't see how the errors are being returned to the plugin caller)